src/ensembl/production/metadata/admin/api/viewsets/dataset.py

```diff
@@ -35,26 +52,68 @@ def get_queryset(self):
+        if not username or not User.objects.filter(username=username).exists():
+            return Response({'detail': 'User not registered'}, status=status.HTTP_401_UNAUTHORIZED)
+
+        # Require Genome_uuid, dataset_type_name, dataset_source
+        genome_uuid = request.data.get('genome_uuid')
+        dataset_type_name = request.data.get('dataset_type')
+        dataset_source_name = request.data.get('dataset_source')
+
+        if not all([genome_uuid, dataset_type_name, dataset_source_name]):
+            return Response({'error': 'Required fields not provided'}, status=status.HTTP_400_BAD_REQUEST)
+        # Check if  Genome_uuid, dataset_type_name, dataset_source are present in the database:
+        if not DatasetSource.objects.filter(name=dataset_source_name).exists():
+            return Response({'error': 'The provided dataset_source_name does not exist in the database'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        try:
+            Genome.objects.get(genome_uuid=uuid.UUID(genome_uuid))
+        except Genome.DoesNotExist:
+            return Response({'error': 'The provided genome_uuid does not exist in the database'},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        if not DatasetType.objects.filter(name=dataset_type_name).exists():
+            return Response({'error': 'The provided dataset_type_name does not exist in the database'},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        # Check if a dateset is already present and not release:
+        existing_datasets = Dataset.objects.filter(
+            dataset_type__name=dataset_type_name,
+            dataset_source__name=dataset_source_name,
+            genome_datasets__genome__genome_uuid=genome_uuid,
+            genome_datasets__release__isnull=True,
+        )
+        if existing_datasets.exists():
+            return Response({
+                'error': 'Dataset with the provided genome_uuid, dataset_type.name and dataset_source.name '
+                         'already exists and is not attached to a release. Please use PUT to update instead of POST.'
+            }, status=status.HTTP_400_BAD_REQUEST)
+
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+    # DELETE:
+    def destroy(self, request, *args, **kwargs):
+        try:
+            instance = self.get_object()
+        except Dataset.DoesNotExist:
+            return Response({'error': 'Dataset not found'}, status=status.HTTP_404_NOT_FOUND)
+
+        try:
+            GenomeDataset.objects.filter(dataset=instance).delete()
+            DatasetAttribute.objects.filter(dataset=instance).delete()
+            if Dataset.objects.filter(dataset_source=instance.dataset_source).count() == 1:
```

Comment:
I believe it's better to request from dataset source (might have less occurrences than dataset)
```suggestion
if DatasetSource.objects.filter(name=instance.dataset_source.name).count() == 1:
```